src/nvim/tui/tui.c

```diff
@@ -294,22 +310,19 @@ static void terminfo_start(UI *ui)
 #endif
 
   // Set up unibilium/terminfo.
-  char *termname = NULL;
+  ui_client_termname = NULL;
   if (term) {
-    os_env_var_lock();
     data->ut = unibi_from_term(term);
-    os_env_var_unlock();
```

Comment:
Can we remove the mutex in env.c now?

Reply:
likely. need to double check the lua threading stuff as well.